File: backend/src/main/java/com/hotel/store/entity/Product.java
package com.hotel.store.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品实体
 */
@Data
@TableName("product")
public class Product {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String name;
    
    private String barcode;
    
    private String category;
    
    private String description;
    
    private BigDecimal price;
    
    private BigDecimal employeePrice;
    
    private Integer stock;
    
    private String image;
    
    private Integer status;
    
    @TableLogic
    private Integer deleted;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/product.ts
import request, { Result } from './request'

export interface Product {
  id: number
  name: string
  description: string
  price: number
  employeePrice?: number
  stock: number
  category: string
  imageUrl?: string
  barcode?: string
}

export const getProductList = () => {
  return request.get<Result<Product[]>>('/product/list')
}

export const getProduct = (id: number) => {
  return request.get<Result<Product>>(`/product/${id}`)
}

export const getProductByBarcode = (barcode: string) => {
  return request.get<Result<Product>>(`/product/barcode/${barcode}`)
}
